### PR TITLE
Support 'labels' parameter for docker.

### LINKED ISSUE #1277: Add support for docker labels
I can come up with a PR, but I already have 2 from March 22 that were not merged yet.

### PR DESCRIPTION
**Pull request by leonty**
This PR fixes #1277 for Ansible v2.x and adds support of the 'label' parameter to the 'docker run' command as it stated in the docker documentation:

```
-l, --label=[]                Set metadata on the container (e.g., --label=com.example.key=value)
```

The new parameter is called 'labels' and has the 'dict' type. Example:

```
    - name: Running a container from new build
      docker: 
        name: "{{ container_name }}"
        image: "{{ container_name }}:{{ build_name }}"
        hostname: "{{ container_name }}.{{ dev_domain }}"
        labels:
          com.mycorp.type: dev
          com.mycorp.name: MyProject
        state: running
```

### CODE REVIEW

**Comment by gregdek** [comment]
@leonty thanks for this PR.

@cove @joshuaconner @softzilla @smashwilson Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

**Comment by AndreiPashkin** [comment]
The same as for #2133, the PR doesn't introduce matching logic for labels. Without it, if you change labels in the playbook, the modules will start a new container, instead changing existing one.

**Comment by leonty** [comment]
Anyone to review this?

File: cloud/docker/docker.py [hunk 1]
```diff
@@ -320,6 +320,13 @@
     default: false
     aliases: []
     version_added: "2.0"
+  labels:
+    description:
+      - Set container labels. Requires docker >= 1.6 and docker-py >= 1.2.0.
+    requered: false
```
Comment (smashwilson) [inline cloud/docker/docker.py]: Spelling: "required"

**Comment by smashwilson** [comment]
@leonty: Other than that tiny spelling touchup this is looking good to me. Give me a ping once you touch that up and I'll give it the go-ahead.

Apologies for the delay and thanks for your patience - life got busy _and_ I stopped using this module heavily in my day job, both of which have impacted my maintainer time heavily :smile:

File: cloud/docker/docker.py [hunk 2]
```diff
@@ -320,6 +320,13 @@
...
+    default: null
+    version_added: "1.9.4"
+
```
Comment (smashwilson) [inline cloud/docker/docker.py]: Would you mind removing the blank line here while you're at it?

**Comment by leonty** [comment]
@smashwilson 
Hi, I've corrected the misspelling and removed an empty line in both PRs.

**Comment by smashwilson** [comment]
:+1:

shipit

**Comment by gregdek** [comment]
Thanks @leonty for the PR and thanks @smashwilson for the review. Marking for potential inclusion.

**Comment by bcoca** [comment]
closed #2133 as dupe, but it seems to also update docker_image, which this one is missing.

**Comment by leonty** [comment]
@bcoca this PR is correct, docker_image should not be touched, the #2133 had a wrong commit from another issue.

**Pull request merged**